src/server/node_manager.c

```diff
@@ -2466,15 +2470,21 @@ discard_job(job *pjob, char *txt, int noack)
 		pn = parse_plus_spec(NULL, &rc);
 	}
 
+	/* Get run vervion of this job */
+	rver = get_jattr_long(pjob, JOB_ATR_run_version);
+
+	if ((mom_superior->mi_dmn_info->dmn_state & INUSE_DOWN) &&
```

Comment:
Pardon my lack of familiarity with discard job, but I'm a bit confused by this check. It seems like discard_job()'s purpose is to have the server(s) send IS_DISCARD_JOB to each mom in case even one of the moms for a job is down. So, even if mother superior is up, it doesn't seem like MS will keep trying to send discard/delete job to the down sister(s) until they come back up, it seems that's the responsibility of the server, not MS. If my understanding is correct, then this check would be incorrect, you'd want the server(s) to ensure that all moms receive IS_DISCARD_JOB, regardless of the MS. Can you please clarify?

Reply:
I could not find a reason why the server is sending the discard to all moms when the MS is up.
If the MS receives discard, it will send delete job to its sisters which will do the purpose. If some of the sisters are down they will get the request from MS when they come up.
So we should remove the case where the server sends the request to all moms when the MS is up. But that can be taken up independently.

Reply:
> I could not find a reason why the server is sending the discard to all moms when the MS is up.
> If the MS receives discard, it will send delete job to its sisters which will do the purpose. If some of the sisters are down they will get the request from MS when they come up.
> So we should remove the case where the server sends the request to all moms when the MS is up. But that can be taken up independently.

There is a reason. It could be that the mother superior is damaged and cannot be up or will late a huge amount of time to get up. If so, the job will remain stuck on the sisters and cannot be deleted, so a force delete makes the server talk to the sisters directly to reclaim the resources blocked by the job.

Reply:
>If some of the sisters are down they will get the request from MS when they come up.

Can you please point me to the code in mom which keeps a track of which sisters haven't responded to delete job request and sends it again when they come up? I couldn't find anything on the mom side to ensure this.

Regardless, I think I agree with Subhasis, it's safer for the server to ensure that discard_job reaches every mom, this is a rare but critical case, so it's important to be correct rather than performant.

Reply:
It's a question of "how long" the resources are blocked. MS could track it, but she herself is down, then the sister resources have no other way of getting cleaned up other than a direct intervention from server. And that intervention is "IS_DISCARD"

Reply:
ya makes sense ..